### PR TITLE
Provide the possibility to give your own httpClient and request configs

### PR DESCRIPTION
**Pull request by bidoubiwa**
Fixes: #1060 
Fixes: #928

### CODE REVIEW

File: packages/instant-meilisearch/__tests__/custom-http-client.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    const documentsTask = await meilisearchClient
+      .index('movies')
+      .addDocuments(dataset)
+    await meilisearchClient.index('movies').waitForTask(documentsTask.taskUid)
+  })
+
+  test('a custom HTTP client', async () => {
+    const httpClient = jest.fn(async (url: string, init?: RequestInit) => {
+      const result = await fetch(url, init)
```
Comment (brunoocasali) [inline packages/instant-meilisearch/__tests__/custom-http-client.test.ts:17]: This is just a wrapper of fetch to create the test, right?

**Review by brunoocasali — APPROVED** [review]
Niceeeeee work 🎉

**Comment by bidoubiwa** [comment]
bors merge

**Comment by bidoubiwa** [comment]
bors try

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**